techniques/system/server-roles/1.0/service-check.st

```diff
@@ -16,64 +16,28 @@
-
       # Raise this class if we are handling a process that should be checked even on
       # relay servers.
       "process_exception_on_relay" expression => strcmp("${check_on_relay_server}", "true");
 
-    any::
-
-      "pass3" expression => "pass2";
-      "pass2" expression => "pass1";
-      "pass1" expression => "any";
-
   methods:
 
-    pass2.(root_server|process_exception_on_relay)::
+    (root_server|process_exception_on_relay)::
 
       "start_process" usebundle => service_ensure_running_path("${initscript}", "${binary}"),
-                     ifvarclass => "!forced_trigger_${canoname}";
-
-      "start_process" usebundle => service_restart("${initscript}"),
-                     ifvarclass => "forced_trigger_${canoname}";
-
-    pass3.(root_server|process_exception_on_relay)::
-
-      "any" usebundle => rudder_common_report("server-roles", "result_success", "&TRACKINGKEY&", "Check ${name} process", "None", "The ${name} process is already running"),
-        ifvarclass => "!process_restart_${canoname}.!forced_trigger_${canoname}";
-
-      "any" usebundle => rudder_common_report("server-roles", "result_repaired", "&TRACKINGKEY&", "Check ${name} process", "None", "The ${name} process was not running and has been restarted"),
-        ifvarclass => "process_restart_${canoname}.service_restart_${c_initscript}_repaired.!forced_trigger_${canoname}";
-
-      "any" usebundle => rudder_common_report("server-roles", "result_repaired", "&TRACKINGKEY&", "Check ${name} process", "None", "The ${name} process has been restarted"),
-        ifvarclass => "service_restart_${c_initscript}_repaired.forced_trigger_${canoname}";
-
-      "any" usebundle => rudder_common_report("server-roles", "result_error", "&TRACKINGKEY&", "Check ${name} process", "None", "The ${name} process couldn't be restarted"),
-        ifvarclass => "service_restart_${c_initscript}_not_ok";
+      "any"           usebundle => rudder_common_reports_generic("server-roles", "service_ensure_running_${c_initscript}", "&TRACKINGKEY&", "Check ${name} process", "None", "Check ${name} process running");
 
-    pass3.(policy_server.!root_server.!process_exception_on_relay)::
+    !(root_server|process_exception_on_relay)::
 
-      "any" usebundle => rudder_common_report("server-roles", "result_success", "&TRACKINGKEY&", "Check ${name} process", "None", "The ${name} process do not need to be checked on relay servers. Skipping...");
-
-  processes:
-
-    root_server|process_exception_on_relay::
-
-    # check the service status
-      "${binary}"
-        comment => "Check the process status",
-        restart_class => "process_restart_${canoname}",
-        classes => kept_if_else("service_running_${canoname}", "service_anomaly_${canoname}", "service_error_${canoname}");
+      "any" usebundle => rudder_common_report("server-roles", "result_na", "&TRACKINGKEY&", "Check ${name} process", "None", "The ${name} process do not need to be checked on relay servers. Skipping...");
```

Comment:
The diff doesn't show this well, but the only change to this line is result_na instead of result_success